DatadogRUM/Sources/Integrations/TelemetryReceiver.swift

```diff
@@ -216,7 +216,7 @@ internal final class TelemetryReceiver: FeatureMessageReceiver {
                 source: .init(rawValue: context.source) ?? .ios,
                 telemetry: .init(
                     message: "[Mobile Metric] \(name)",
-                    telemetryInfo: attributes.enrichIfNeeded(with: context)
+                    telemetryInfo: attributes.enrichWithDeviceOsInfo(using: context)
```

Comment:
change-request/ We did this change recently: [**RUM-1844: Add device and os info to telemetry events**](https://github.com/DataDog/rum-events-format/pull/200) 💡. Both `os` and `device` attributes are now part of telemetry model.

We should update RUM models and use type-safe attributes rather than attaching `os` and `device` info through generic `telemetryInfo` (which is `[String: Encodable]`)